File: ean0354_lab01.py
# Ethyn Nguyen
# 1001930354
# Python
# linux ubuntu

# Omega: python ean0354_lab01.py

import os

# Returns the total size in bytes of all files in the current directory / folder and all sub-folders
def dir_size(path = "."):
    total_size = 0
    for file in os.listdir(path): # Continues to loop until end of directory
        full_path = os.path.join(path, file)
        if os.path.isdir(full_path):
            total_size += dir_size(full_path) # Recursively go through each directory
        else:
            total_size += os.path.getsize(full_path) # Add file size to total_size
            
    return total_size

print(dir_size(".")) # Call to function

File: ean0354_lab01.java
// Ethyn Nguyen
// 1001930354
// Java
// linux ubuntu

// Omega: javac ean0354_lab01.java

import java.io.File;

public class ean0354_lab01 {
    private int dirSize(File directory) {
        int totalSize = 0;
       
        // listFiles() returns an array of pathnames denoting the files in the directory
        File[] files = directory.listFiles();
 
        // Continues to loop until end of directory
        for (File file : files) {
            if (file.isDirectory())
                totalSize += dirSize(file); // Recursively go through each directory
            else
                totalSize += file.length(); // Add file size to totalSize
        }
        return totalSize;
    }

    public static void main(String[] args) {
        ean0354_lab01 lab = new ean0354_lab01(); // Class instance

        int bytes = lab.dirSize(new File(".")); // Total amount of directory size in bytes
 
        System.out.println(bytes);
    }
}
